src/web/mjs/connectors/CxC.mjs

```diff
@@ -0,0 +1,254 @@
+            [].reduce.call(
+                new Uint8Array(encryptedImage),
+                function (e, _) {
+                    return e + String.fromCharCode(_);
+                },
+                ''
+            )
+        );
+        const token = this.auth.accessToken ? this.auth.accessToken : 'freeforcxc2021reading'; //default free content token, may change in the future
+        const tokenHash = CryptoJS.SHA512(token).toString();
+        const tokenCombo = {
+            key: tokenHash.substr(0, 64),
+            iv: tokenHash.substr(30, 32)
+        };
+        let imageComboString = this._decrypt(payload.imageKey, tokenCombo).split(':');
+        const imageCombo = {
+            key: imageComboString[0],
+            iv: imageComboString[1]
+        };
+        let decryptedImageResponse = this._decrypt(encTransformImage, imageCombo).split(",");
+        // from https://cxc.today/worker/base64ToBlob.js
+        const e = decryptedImageResponse[0].indexOf("base64") >= 0 ? atob(decryptedImageResponse[1]) : decodeURI(decryptedImageResponse[1]);
+        let n = new Uint8Array(e.length);
+        for (let i = 0; i < e.length; i++) {
+            n[i] = e.charCodeAt(i);
+        }
+        return {
+            mimeType: `image/${payload.url.split('.').pop()}`,
+            data: n
+        };
+    }
+
+    _decrypt(encryptedBuffer, keyIvCombo) {
+        // from https://cxc.today/worker/AESDecrypt.js
+        let decryptedBuffer = CryptoJS.AES.decrypt(encryptedBuffer, CryptoJS.enc.Hex.parse(keyIvCombo.key), {
+            iv: CryptoJS.enc.Hex.parse(keyIvCombo.iv),
+            mode: CryptoJS.mode.CBC,
+            padding: CryptoJS.pad.Pkcs7
+        });
+        return decryptedBuffer.toString(CryptoJS.enc.Utf8);
+    }
+
+    async _getUUID() {
+        this.uuid = await this._getStorageValue('uuid');
+        if (!this.uuid) {
+            let request = new Request(`${this.api}/guest`, this.requestOptions);
+            let data = await this.fetchJSON(request);
+            this.uuid = data.data;
+        }
+        this.requestOptions.headers.set('uuid', this.uuid);
+        return;
+    }
+
+    async _login() {
+        if (this.config.username.value && this.config.password.value && !this.auth.accessToken) {
+            this.auth.accessToken = await this._getStorageValue('accessToken');
+            this.auth.refreshToken = await this._getStorageValue('refreshToken');
+            if (!this.auth.accessToken) {
+                try {
+                    let loginRequest = new Request(`${this.api}/auth/login`, {
+                        credentials: 'omit',
+                        method: 'POST',
+                        body: JSON.stringify({
+                            grant_type: 'password',
+                            password: this.config.password.value,
+                            username: this.config.username.value
+                        }),
+                        headers: this.requestOptions.headers
```

Comment:
Since all requests besides the one to get the image are for JSON, and not having the JSON headers can result in a 400 for some endpoints, I don't think it makes sense to add it for every request that isn't for the image rather than exclude it for just the image request.